File: src/api/codecheck/exceptions/base_exception.py
# -*- coding: utf-8 -*-
"""
Tencent is pleased to support the open source community by making BK-BASE 蓝鲸基础计算平台 available.
Copyright (C) 2019 THL A29 Limited, a Tencent company. All rights reserved.
Licensed under the MIT License (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at http://opensource.org/licenses/MIT
Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and limitations under the License.
"""

from common.errorcodes import ErrorCode
from common.exceptions import BaseAPIError
from django.utils.translation import ugettext_lazy as _


class CodeCheckCommonCode(object):
    # 参数异常
    ILLEGAL_ARGUMENT_EX = ("001", _(u"参数异常"))
    # 业务逻辑问题
    LOGIC_ERR = ("100", _(u"代码校验异常"))
    # 期望外异常
    UNEXPECT_EX = ("200", _(u"预期外异常"))


class CommonException(BaseAPIError):
    """
    Batch Base class for all batch exceptions.
    """

    MODULE_CODE = ErrorCode.BKDATA_DATAAPI_CODECHECK

    def __init__(self, message, code=000):
        self.CODE = code
        self.MESSAGE = message
        super(CommonException, self).__init__()

File: src/dataflow/unified-computing/batch/batch-common/src/main/scala/com/tencent/bk/base/dataflow/spark/exception/BaseException.scala
/*
 * Tencent is pleased to support the open source community by making BK-BASE 蓝鲸基础平台 available.
 *
 * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
 *
 * BK-BASE 蓝鲸基础平台 is licensed under the MIT License.
 *
 * License for BK-BASE 蓝鲸基础平台:
 * --------------------------------------------------------------------
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
 * and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT
 * LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
 * NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
 * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
package com.tencent.bk.base.dataflow.spark.exception

import com.tencent.bk.base.dataflow.spark.exception.code.{CodePrefix, ErrorCode}
import com.tencent.bk.base.dataflow.spark.logger.MDCLogger
import MDCLogger.Level

/**
  * 基础异常
  */
abstract class BaseException(message: String) extends RuntimeException(message) {
  def getCodePrefix(): CodePrefix

  @deprecated
  def this(message: String, cause: Option[Throwable], errCode: Option[String]) {
    this(s"${errCode.getOrElse(ErrorCode.DEFAULT)} -> ${message}")
    cause match {
      case Some(e: Throwable) => initCause(e)
      case _ =>
    }
    errCode match {
      case Some(code: String) =>
        MDCLogger.logger(message, errCode = s"${getCodePrefix()}${code}", level = Level.ERROR)
      case _ =>
    }
  }

  def this(message: String, cause: Throwable, errCode: String) {
    this(message, Option(cause), Option(errCode))
  }

  def this(message: String, errCode: String) {
    this(message, None, Option(errCode))
  }

  @deprecated
  def this(message: String, cause: Throwable) {
    this(message, Option(cause), None)
  }

  @deprecated
  def this(cause: Throwable) {
    this(Option(cause).map(_.toString).orNull, cause)
  }

  @deprecated
  def this() {
    this("", None, None)
  }
}
